src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php

```diff
@@ -0,0 +1,233 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Sell\Catalog;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Category\Command\ToggleCategoryStatusCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Category\Command\UpdateCategoriesStatusCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Category\Exception\CannotUpdateCategoryStatusException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Category\Exception\CategoryException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Category\Exception\CategoryNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Category\ValueObject\CategoryId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Category\ValueObject\CategoryStatus;
+use PrestaShop\PrestaShop\Core\Search\Filters\CategoryFilters;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class CategoryController is responsible for "Sell > Catalog > Categories" page.
+ */
+class CategoryController extends FrameworkBundleAdminController
+{
+    /**
+     * Show categories listing.
+     *
+     * @param Request $request
+     * @param CategoryFilters $filters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, CategoryFilters $filters)
+    {
+        $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
+        $categoryGridFactory = $this->get('prestashop.core.grid.factory.category');
+        $categoryGrid = $categoryGridFactory->getGrid($filters);
+
+        $categoriesKpiFactory = $this->get('prestashop.core.kpi_row.factory.categories');
+
+        $currentCategoryId = $filters->getFilters()['id_category_parent'];
+        $categoryViewDataProvider = $this->get('prestashop.adapter.category.category_view_data_provider');
+        $categoryViewData = $categoryViewDataProvider->getViewData($currentCategoryId);
```

Comment:
i think `GetCategoryForListViewQuery` or similar should be used instead of `CategoryViewDataProvider` adapter? @eternoendless what do you think? 

Use case:
Once you open Categories list, it resolves parent category for which list of child categories are displayed. By default, it uses Home category and displays all of it children. So currently `CategoryViewDataProvider:getViewData($parentCategoryId)` returns this kind of data:
```php
return [
    'breadcrumb_tree' => $categoriesTree, // used to display breadrumb
    'id' => $category->id,
    'id_parent' => $category->id_parent,
    'is_home_category' => $this->configuration->get('PS_HOME_CATEGORY') == $category->id,
    'name' => $category->name[$this->contextLangId], // used to display layout title
];
```
Im thinking that Query could return some DTO with this data instead of array.

Reply:
note for any reviewer: get the pull request and try on your own computer, this grid is a little bit different from what we use :confused: